From 3e22d22eadb459cb865bd96bc363bfef00c75867 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 14 Jun 2016 10:30:58 +0300
Subject: [PATCH 1196/1240] fix: clk: ap806: fix clock MSS calculation for
 AP-806-A0

Update soc_mss_clk_get function to return the correct clock
for AP-806-A0 (fixed clock 200MHz)
- soc_mss_clk_get used only in U-Boot clock print

Change-Id: I89fc0b0a7cddc70f2676ef47681df9f402d29490
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30445
Reviewed-by: Haim Boot <hayim@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/cpu/armv8/armada8k/clock.c | 10 ++++------
 include/configs/armada8k.h          |  2 +-
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada8k/clock.c b/arch/arm/cpu/armv8/armada8k/clock.c
index d7a563d..cde65fe 100644
--- a/arch/arm/cpu/armv8/armada8k/clock.c
+++ b/arch/arm/cpu/armv8/armada8k/clock.c
@@ -34,18 +34,16 @@ u32 soc_ring_clk_get(void)
 
 u32 soc_mss_clk_get(void)
 {
+#ifdef CONFIG_AP806_Z_SUPPORT
 	return soc_ring_clk_get() / MSS_CLOCK_DIV;
+#else
+	return CONFIG_MSS_FREQUENCY;
+#endif
 }
 
 u32 soc_tclk_get(void)
 {
-#ifdef CONFIG_PALLADIUM
-	return CONFIG_MSS_FREQUENCY;
-#elif defined CONFIG_AP806_Z_SUPPORT
 	return soc_mss_clk_get();
-#else
-	return CONFIG_TCLK_FREQUENCY;
-#endif
 }
 
 
diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 887f514..4f821e2 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -47,9 +47,9 @@
 #define CONFIG_REMAKE_ELF
 
 /* Clock Definitions */
-#define CONFIG_TCLK_FREQUENCY	200 * MHz
 #ifndef CONFIG_PALLADIUM
 #define COUNTER_FREQUENCY	(25 * 1000000)
+#define CONFIG_MSS_FREQUENCY	(200 * 1000000)
 #else
 #define COUNTER_FREQUENCY	(48000)
 #define CONFIG_MSS_FREQUENCY	(384000)
-- 
1.9.1

